[BUG] Flakey test failure in SortedSetDVStringFieldDataTests testSortMissingLast

**Describe the bug**

First observed in https://github.com/opensearch-project/OpenSearch/pull/4183#issuecomment-1215586520, but I hit the same failure several times in #4222 and was able to reproduce it locally:

```
REPRODUCE WITH: ./gradlew ':server:test' --tests "org.opensearch.index.fielddata.SortedSetDVStringFieldDataTests.testSortMissingLast" -Dtests.seed=BC7C1431A4F70C5D -Dtests.security.manager=true -Dtests.jvm.argline="-XX:TieredStopAtLevel=1 -XX:ReservedCodeCacheSize=64m" -Dtests.locale=ga-IE -Dtests.timezone=Europe/Belgrade -Druntime.java=17

org.opensearch.index.fielddata.SortedSetDVStringFieldDataTests > testSortMissingLast FAILED
    java.lang.AssertionError: expected:<1105> but was:<1023>
```

**To Reproduce**
See above

**Expected behavior**
Test consistently passes

**Plugins**
None

**Screenshots**
N/A

**Host/Environment (please complete the following information):**
 - OS: [e.g. iOS]
 - Version [e.g. 22]

**Additional context**

--- a/server/src/test/java/org/opensearch/index/fielddata/AbstractStringFieldDataTestCase.java
+++ b/server/src/test/java/org/opensearch/index/fielddata/AbstractStringFieldDataTestCase.java
@@ -52,6 +52,7 @@ import org.apache.lucene.search.Sort;
 import org.apache.lucene.search.SortField;
 import org.apache.lucene.search.TermQuery;
 import org.apache.lucene.search.TopFieldDocs;
+import org.apache.lucene.search.TotalHits;
 import org.apache.lucene.search.join.QueryBitSetProducer;
 import org.apache.lucene.search.join.ScoreMode;
 import org.apache.lucene.search.join.ToParentBlockJoinQuery;
@@ -340,7 +341,12 @@ public abstract class AbstractStringFieldDataTestCase extends AbstractFieldDataI
             randomBoolean() ? numDocs : randomIntBetween(10, numDocs),
             new Sort(sortField)
         );
-        assertEquals(numDocs, topDocs.totalHits.value);
+        // As of Lucene 9.0.0, totalHits may be a lower bound
+        if (topDocs.totalHits.relation == TotalHits.Relation.EQUAL_TO) {
+            assertEquals(numDocs, topDocs.totalHits.value);            
+        } else {
+            assertTrue(numDocs >= topDocs.totalHits.value);                        
+        }
         BytesRef previousValue = first ? null : reverse ? UnicodeUtil.BIG_TERM : new BytesRef();
         for (int i = 0; i < topDocs.scoreDocs.length; ++i) {
             final String docValue = searcher.doc(topDocs.scoreDocs[i].doc).get("value");

--- a/server/src/test/java/org/opensearch/index/fielddata/AbstractStringFieldDataTestCase.java
+++ b/server/src/test/java/org/opensearch/index/fielddata/AbstractStringFieldDataTestCase.java
@@ -343,9 +343,10 @@ public abstract class AbstractStringFieldDataTestCase extends AbstractFieldDataI
         );
         // As of Lucene 9.0.0, totalHits may be a lower bound
         if (topDocs.totalHits.relation == TotalHits.Relation.EQUAL_TO) {
-            assertEquals(numDocs, topDocs.totalHits.value);            
+            assertEquals(numDocs, topDocs.totalHits.value);
         } else {
-            assertTrue(numDocs >= topDocs.totalHits.value);                        
+            assertTrue(1000 <= topDocs.totalHits.value);
+            assertTrue(numDocs >= topDocs.totalHits.value);
         }
         BytesRef previousValue = first ? null : reverse ? UnicodeUtil.BIG_TERM : new BytesRef();
         for (int i = 0; i < topDocs.scoreDocs.length; ++i) {

